front/src/modules/timesStops/helpers/utils.ts

```diff
@@ -125,3 +139,145 @@ export function disabledTextColumn(
     disabled: true,
   };
 }
+
+/**
+ * Synchronizes arrival, departure and stop times.
+ * updates onStopSignal
+ * updates isMarginValid and theoreticalMargin
+ */
+export function updateRowTimesAndMargin(
+  rowData: PathWaypointRow,
+  previousRowData: PathWaypointRow,
+  op: { fromRowIndex: number },
+  allWaypointsLength: number
+): PathWaypointRow {
+  const newRowData = { ...rowData };
+  if (
+    newRowData.departure &&
+    newRowData.arrival &&
+    (!isEqual(newRowData.arrival, previousRowData.arrival) ||
+      !isEqual(newRowData.departure, previousRowData.departure))
+  ) {
+    newRowData.stopFor = String(
+      durationInSeconds(
+        time2sec(newRowData.arrival.time || ''),
+        time2sec(newRowData.departure.time || '')
+      )
+    );
+  }
+  if (!newRowData.stopFor && op.fromRowIndex !== allWaypointsLength - 1) {
+    newRowData.onStopSignal = false;
+  }
+  newRowData.isMarginValid = !(
+    newRowData.theoreticalMargin && !marginRegExValidation.test(newRowData.theoreticalMargin)
+  );
+  if (newRowData.isMarginValid && op.fromRowIndex === 0) {
+    newRowData.arrival = undefined;
+    // As we put 0% by default for origin's margin, if the user removes a margin without
+    // replacing it to 0% (undefined), we change it to 0%
+    if (!newRowData.theoreticalMargin) {
+      newRowData.theoreticalMargin = '0%';
+    }
+  }
```

Comment:
I have a weird thing going one with margins. If I enter "10" it doesn't update the store and it's not displayed as "invalid" (in red). The store is updated only if I enter a valid value.

[Capture vidéo du 11-09-2024 17:58:15.webm](https://github.com/user-attachments/assets/8885f0f7-0832-470f-9403-0f9482f60a48)

Reply:
I think the original design decision to only put valid values in the store shows its limits.
I fixed it to update locally.
When you put an invalid value, it only updates `rows` locally,  but when we update another cell that’s valid, it’s going to be update the entire table from the new data from the store. as the invalid cell was not in store, it’s going to get erase.
but it might be an acceptable behavior.

@axrolld we can check that together ?

Reply:
Thank you for the change. The current behavior seems ok to me.